aten/src/ATen/preprocess_declarations.py

```diff
@@ -76,9 +76,10 @@ def expand(types):
         if 'CPU' in backend_types:
             backend_types['CPU'].discard('BFloat16')
 
-    # TODO: remove this hack once support for a bfloat16 tensor for CUDA is enabled
-    if 'CUDA' in backend_types:
-        backend_types['CUDA'].discard('BFloat16')
+    # special case remove BFloat16 for cuda unless it is explicitly enabled
+    if not option.get('cuda_bfloat16', False):
+        if 'CUDA' in backend_types:
+            backend_types['CUDA'].discard('BFloat16')
```

Comment:
This needs to be removed eventually, ROCm overloads CUDA.

Reply:
Shall I remove this check completely in this PR or keep it as is and remove it when ops are enabled?